java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsClusterCommands.java

```diff
@@ -0,0 +1,81 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.commands;
+
+import glide.api.models.configuration.RequestRoutingConfiguration.Route;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Supports commands and transactions for the "Scripting and Function" group for a cluster client.
+ *
+ * @see <a href="https://redis.io/docs/latest/commands/?group=scripting">Scripting and Function
+ *     Commands</a>
+ */
+public interface ScriptingAndFunctionsClusterCommands {
+
+    /**
+     * Loads a library to Redis.<br>
+     * The command will be routed to all primary nodes.
+     *
+     * @see <a href="https://redis.io/docs/latest/commands/function-load/">redis.io</a> for details.
+     * @param libraryCode The source code that implements the library.
+     * @return The library name that was loaded.
+     * @example
+     *     <pre>{@code
+     * String code = "#!lua name=mylib \n redis.register_function('myfunc', function(keys, args) return args[1] end)";
+     * String response = client.functionLoad(code).get();
+     * assert response.equals("mylib");
+     * }</pre>
+     */
+    CompletableFuture<String> functionLoad(String libraryCode);
+
+    /**
+     * Loads a library to Redis and overwrites the existing library with the new contents.<br>
+     * The command will be routed to all primary nodes.
+     *
+     * @see <a href="https://redis.io/docs/latest/commands/function-load/">redis.io</a> for details.
+     * @param libraryCode The source code that implements the library.
+     * @return The library name that was loaded.
+     * @example
+     *     <pre>{@code
+     * String code = "#!lua name=mylib \n redis.register_function('myfunc', function(keys, args) return args[1] end)";
+     * String response = client.functionLoadWithReplace(code).get();
+     * assert response.equals("mylib");
+     * }</pre>
+     */
+    CompletableFuture<String> functionLoadWithReplace(String libraryCode);
+
+    /**
+     * Loads a library to Redis.
+     *
+     * @see <a href="https://redis.io/docs/latest/commands/function-load/">redis.io</a> for details.
+     * @param libraryCode The source code that implements the library.
+     * @param route Specifies the routing configuration for the command. The client will route the
+     *     command to the nodes defined by <code>route</code>.
+     * @return The library name that was loaded.
+     * @example
+     *     <pre>{@code
+     * String code = "#!lua name=mylib \n redis.register_function('myfunc', function(keys, args) return args[1] end)";
+     * Route route = new SlotKeyRoute("key", PRIMARY);
+     * String response = client.functionLoad(code, route).get();
+     * assert response.equals("mylib");
+     * }</pre>
+     */
+    CompletableFuture<String> functionLoad(String libraryCode, Route route);
```

Comment:
Do we want this?  Something to ask Bar. 
Unless we know for certain which node the script should be loaded to (based on key) I'm figuring functions and scripts should be loaded to all primary nodes.

Reply:
This is why invoke script doesn't take a route.

Reply:
I based on routing definitions given in `redis-rs`.

Reply:
> As noted above, Redis automatically handles propagation of loaded functions to replicas. In a Redis Cluster, it is also necessary to load functions to all cluster nodes. This is not handled automatically by Redis Cluster, and needs to be handled by the cluster administrator (like module loading, configuration setting, etc.).